dotCMS/src/main/java/com/dotcms/saml/DotIdentityProviderConfigurationImpl.java

```diff
@@ -0,0 +1,130 @@
+package com.dotcms.saml;
+
+import com.dotcms.security.apps.AppSecrets;
+import com.dotcms.security.apps.AppsAPI;
+import com.dotcms.security.apps.Secret;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+
+import java.util.Optional;
+
+/**
+ * Default implementation to retrieve the configuration from apps
+ * @author jsanca
+ */
+public class DotIdentityProviderConfigurationImpl implements IdentityProviderConfiguration {
+
+    private final AppsAPI    appsAPI;
```

Comment:
Codacy found an issue: [Perhaps 'appsAPI' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5707914)